src/kOS.Safe/Execution/YieldFinishedCompile.cs

```diff
@@ -40,7 +40,8 @@ private YieldFinishedCompile(GlobalPath scriptPath, int lineNumber, string fileC
 
         public override void ThreadInitialize(SafeSharedObjects shared)
         {
-            programContext = shared.Cpu.SwitchToProgramContext();
+            if (compileMode != CompileMode.FILE)
+                programContext = shared.Cpu.SwitchToProgramContext(); // only switch the context if executing
```

Comment:
This was a good catch.  Yup, this is exactly what was the problem.  This tries to run the program... that isn't being made.